connect/transforms/src/main/java/org/apache/kafka/connect/transforms/TimestampRouter.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.errors.DataException;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * This transformation facilitates updating the record's topic field as a function of the original topic value and the record timestamp.
+ * <p/>
+ * The topic field determines what topic the records from a source connector are routed to.
+ * In the case of a sink connector, the topic field is often used to determine the equivalent entity name in the destination system (e.g. database table name).
+ */
+public class TimestampRouter<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final ConfigDef CONFIG_DEF = new ConfigDef()
+            .define("topic.format", ConfigDef.Type.STRING, "${topic}-${timestamp}", ConfigDef.Importance.HIGH,
```

Comment:
Hmm, will this be a good set of defaults? I'm actually questioning how useful this transformation is since you'd have to create new topics. The defaults seem pretty dangerous since it'll create a new topic every day. Maybe this really only makes sense for sink connectors but happens to be easy to support both source and sink?